analysis/org.eclipse.tracecompass.analysis.timing.core/src/org/eclipse/tracecompass/internal/analysis/timing/core/segmentstore/SegmentStoreTreeStatisticsDataProvider.java

```diff
@@ -0,0 +1,180 @@
+/**********************************************************************
+ * Copyright (c) 2024, 2026 Ericsson
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License 2.0 which
+ * accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+package org.eclipse.tracecompass.internal.analysis.timing.core.segmentstore;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.tracecompass.analysis.timing.core.segmentstore.SegmentStoreStatisticsModel;
+import org.eclipse.tracecompass.analysis.timing.core.statistics.ITreeStatistics;
+import org.eclipse.tracecompass.analysis.timing.core.statistics.ITreeStatisticsAnalysis;
+import org.eclipse.tracecompass.common.core.NonNullUtils;
+import org.eclipse.tracecompass.internal.analysis.timing.core.segmentstore.SegmentStoreStatisticsAspects.NamedStatistics;
+import org.eclipse.tracecompass.internal.tmf.core.model.filters.FetchParametersUtils;
+import org.eclipse.tracecompass.segmentstore.core.ISegment;
+import org.eclipse.tracecompass.tmf.core.analysis.IAnalysisModule;
+import org.eclipse.tracecompass.tmf.core.dataprovider.DataProviderParameterUtils;
+import org.eclipse.tracecompass.tmf.core.model.CommonStatusMessage;
+import org.eclipse.tracecompass.tmf.core.model.filters.FilterTimeQueryFilter;
+import org.eclipse.tracecompass.tmf.core.model.filters.TimeQueryFilter;
+import org.eclipse.tracecompass.tmf.core.model.tree.TmfTreeModel;
+import org.eclipse.tracecompass.tmf.core.response.ITmfResponse.Status;
+import org.eclipse.tracecompass.tmf.core.response.TmfModelResponse;
+import org.eclipse.tracecompass.tmf.core.trace.ITmfTrace;
+
+/**
+ * This data provider supplies a statistics tree for analysis modules
+ * implementing {@link ITreeStatisticsAnalysis}, offering a hierarchical
+ * structure of statistics across multiple levels. By passing a
+ * {@link FilterTimeQueryFilter}, it also returns statistics for the
+ * specified range.
+ *
+ * @author Siwei Zhang
+ * @since 4.0
+ */
+public class SegmentStoreTreeStatisticsDataProvider extends AbstractSegmentStoreStatisticsDataProvider {
+
+    /**
+     * Base {@link SegmentStoreTreeStatisticsDataProvider} prefix
+     */
+    public static final String ID = "org.eclipse.tracecompass.analysis.timing.core.segmentstore.SegmentStoreTreeStatisticsDataProvider"; //$NON-NLS-1$
+
+    private final ITreeStatisticsAnalysis fProvider;
+    private final @Nullable IAnalysisModule fModule;
+
+    private final Map<String, Long> fIdToType = new HashMap<>();
```

Comment:
This map is not needed here because it's in the base class. Please remove and update also the dispose() method for that.